src/window/mod.rs

```diff
@@ -143,7 +147,16 @@ pub fn create_window() {
 
     let mut update_loop = UpdateLoop::new();
 
-    event_loop.run(move |e, _window_target, control_flow| {
-        *control_flow = update_loop.step(&mut window_wrapper, e);
-    });
+    match event_loop.run(move |e, _window_target, control_flow| {
+        if e != Event::LoopExiting {
+            *control_flow = update_loop.step(&mut window_wrapper, e);
+        }
+    }) {
+        Ok(()) => 0,
+        // All error codes have to be u8, so just do a direct cast with wrap around, even if the value is negative,
+        // that's how it's normally done
```

Comment:
Why do all error codes have to be `u8`? Is consistency with errors on Unix vs. Windows from concern if internals error out?

Reply:
Since Unix really only support U8, rust made the decision to only support that. https://doc.rust-lang.org/stable/std/process/struct.ExitCode.html

We could change all our internal uses to u8, but unfortunately winit uses integer exit codes, and the exit codes we receive from Neovim are also integers. So the most straightforward approach is to use int32 everywhere and convert it to u8 just before returning it.

Reply:
The docs do mention https://doc.rust-lang.org/std/process/fn.exit.html as a usable alternative, we could call that as the last line in main.

Reply:
The problem is that it does not call any destructors, so we will not properly wait for Neovim to terminate for example. Of course, it's possible to wrap the function to ensure that, but I prefer a totally clean exit if possible. 

https://doc.rust-lang.org/std/process/fn.exit.html